src/models/tools/geometry/geometry-export.ts

```diff
@@ -0,0 +1,370 @@
+    let props: any = {};
+    _changes.forEach(change => {
+      props = {...props, ...change.properties };
+    });
+    if (props.id !== id) props.id = id;
+    const type = `"type": "movableLine"`;
+    const pointIds = getMovableLinePointIds(id);
+    const { parents: pt0Position } = getPointExportables(pointIds[0]);
+    const { parents: pt1Position } = getPointExportables(pointIds[1]);
+    const parents = `"parents": [${exportMovableLinePoint(pt0Position)}, ${exportMovableLinePoint(pt1Position)}]`;
+    const otherProps = Object.keys(props).length > 0
+                        ? `"properties": ${JSON.stringify(props)}`
+                        : "";
+    return `{ ${type}, ${parents}${comma(!!otherProps)}${otherProps} }${comma(!isLast)}`;
+  };
+
+  const exportFnMap: Partial<Record<JXGObjectType, (id: string, isLast: boolean) => string>> = {
+    image: exportImage,
+    movableLine: exportMovableLine,
+    point: exportPoint,
+    polygon: exportPolygon,
+    vertexAngle: exportVertexAngle
+  };
+
+  const exportObject = (id: string, isLast: boolean) => {
+    const type = objectInfoMap[id]?.type;
+    return type && exportFnMap[type]?.(id, isLast);
+  };
+
+  const exportObjects = () => {
+    pushLine(`"objects": [`, 2);
+    let lastExportedId: string;
+    orderedIds.forEach(id => {
+      if (isExportable(id)) {
+        lastExportedId = id;
+      }
+    });
+    orderedIds.forEach(id => {
+      if (isExportable(id)) {
+        const objLine = exportObject(id, id === lastExportedId);
+        objLine && castArray(objLine).forEach(line => pushLine(line, 4));
+      }
+    });
+    pushLine(`]`, 2);
+  };
+
+  // loop through each change, adding it to the set of changes that affect each object
+  changes.forEach(changeJson => {
+    const change = safeJsonParse<JXGChange>(changeJson);
+    if (change) {
+      const ids = getTargetIdsFromChange(change);
+      const dependencies = getDependenciesFromChange(change, objectInfoMap);
+      ids.forEach(id => {
+        // track movable line control points so we can track changes to them
+        if ((change.operation === "create") && (change.target === "movableLine")) {
+          getMovableLinePointIds(id).forEach(ptId => {
+            objectInfoMap[ptId] = {
+              id: ptId,
+              type: "point",
+              changes: [change],
+              dependents: [],
+              dependencies: [],
+              noExport: true  // movable line control points aren't individually exportable
+            };
+            orderedIds.push(ptId);
+          });
+        }
+        if (!objectInfoMap[id]) {
+          objectInfoMap[id] = {
+            id,
+            type: change.target,
+            changes: [change],
+            dependents: [],
+            dependencies
+          };
+          dependencies.forEach(independentId => {
+            const objectInfo = objectInfoMap[independentId];
+            objectInfo.dependents.push(id);
```

Comment:
I see an occasional exception here due to `objectInfo` being undefined.  Do we need to add protection around the push?

Reply:
I added a commit which handles the linked tables/points.